### PR TITLE
Add a Matrix of Linux-based Build Combinations to Travis CI

### PR DESCRIPTION
**Pull request by gerickson**

### CODE REVIEW

File: test/test-oneshot-timer.c [hunk 1]
```diff
@@ -167,7 +172,7 @@ int main(int argc, char **argv)
 #endif
 
     NL_LOG_CRIT(lrTEST, "start main (after initializing runtime)\n");
-
+]
```
Comment (aiw-google) [inline test/test-oneshot-timer.c]: I don't think you want that.

Comment (gerickson) [reply test/test-oneshot-timer.c]: Thanks, fixed and pushed.

File: test/test-settings.c [hunk 2]
```diff
@@ -25,104 +25,152 @@
...
-        int err;
+        int status;
 
-        err = nl_settings_get_value_as_int(idx % nl_settings_keyMax, &oldvalue);
+        status = nl_settings_get_value_as_int(idx % nl_settings_keyMax, &oldvalue);
+        NL_LOG_CRIT(lrTEST, "get value result: %d\n", status);
+        NLER_ASSERT(status == NLER_SUCCESS);
 
-        if (err == NLER_SUCCESS)
+        if (status == NLER_SUCCESS)
```
Comment (aiw-google) [inline test/test-settings.c]: NLER_ASSERT behaves like nlABORT, not nlASSERT: It's always enabled, regardless of the state of NDEBUG or any other constant.  So its unnecessary to have two checks for status == NLER_SUCCESS.

File: test/test-settings.c [hunk 3]
```diff
@@ -188,64 +237,145 @@ void taskEntryB(void *aParams)
...
-    int             idx;
-    int             status;
-    int             retval;
-    nl_event_t      *queuemem[50];
-    nleventqueue_t  queue;
+    bool retval = false;
 
-    NL_LOG_CRIT(lrTEST, "start main\n");
+    if ((!aSubscriber->mFailed && !aPublisher->mFailed) &&
+        (!aSubscriber->mSucceeded && !aPublisher->mSucceeded))
```
Comment (aiw-google) [inline test/test-settings.c]: If any one of those four is true, we return false.  Is that what's desired here, or do we want something like this:

    if ((!aSubscriber->mFailed && !aSubscriber->mSucceeded) ||
        (!aPublisher->mFailed  && !aPublisher->mSucceeded))

File: test/test-settings.c [hunk 4]
```diff
@@ -188,64 +237,145 @@ void taskEntryB(void *aParams)
...
+                           volatile const taskData_t *aPublisher)
+{
+    bool retval = false;
 
-    status = nleventqueue_create(queuemem, sizeof(queuemem), &queue);
-    NLER_ASSERT(status == NLER_SUCCESS);
+    if (aSubscriber->mFailed || aPublisher->mFailed)
+        retval = false;
+    else if (aSubscriber->mSucceeded && aPublisher->mSucceeded)
+        retval = true;
```
Comment (aiw-google) [inline test/test-settings.c]: Missing braces.

File: test/Makefile.am [hunk 5]
```diff
@@ -83,19 +84,33 @@ check_PROGRAMS                                += \
...
 if !NLER_BUILD_EVENT_TIMER
 check_PROGRAMS                                += \
     test-subpub                                  \
     test-timer                                   \
     $(NULL)
 endif # !NLER_BUILD_EVENT_TIMER
 
+# Test applications that should be built but not installed and not run
+# when the 'check' target is run but should always be built to ensure
+# overall "build sanity".
```
Comment (aiw-google) [inline test/Makefile.am]: Maybe a cut and paste error here?  Looks like this should be something like:

    Applications that should not be installed or run when the 'check' target is run,
    but which should always be built to ensure overall "build sanity".

**Review by aiw-google — APPROVED** [review]
LGTM, but see my comments.

File: test/test-settings.c (hunk 3 above)
Comment (gerickson) [reply test/test-settings.c]: I think there is a bug there; however, I believe this is what is actually desired:

/**
 *  Determine whether or not the main thread should continue to wait
 *  for the test to complete.
 *
 *  The main thread should wait for testing until:
 *
 *    * Either thread failed (mFailed == true).
 *    * Both threads succeeded (mSucceeded == true).
 *
 */
static bool is_testing(volatile const taskData_t *aSubscriber,
                       volatile const taskData_t *aPublisher)
{
    bool retval = true;

    if ((aSubscriber->mFailed || aPublisher->mFailed) ||
        (aSubscriber->mSucceeded && aPublisher->mSucceeded))
    {
        retval = false;
    }

    return retval;
}

File: test/test-settings.c (hunk 2 above)
Comment (gerickson) [reply test/test-settings.c]: Thanks, fixed.

File: test/test-settings.c (hunk 3 above)
Comment (gerickson) [reply test/test-settings.c]: Fixed and pushed.

File: test/test-settings.c (hunk 4 above)
Comment (gerickson) [reply test/test-settings.c]: Thanks, fixed.

File: test/Makefile.am (hunk 5 above)
Comment (gerickson) [reply test/Makefile.am]: Thanks, I took another pass at this and pushed it.

**Pull request merged**